react-app/src/app/screens/Game/index.js

```diff
@@ -1,58 +1,48 @@
 import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { jumpTo, toggleMark, updateStatus } from '../../../redux/Game/actions';
 
 import calculateWinner from './utils';
 import Board from './components/Board';
 import style from './styles.scss';
 
 class Game extends Component {
-  state = {
-    history: [{ squares: Array(9).fill(null) }],
-    stepNumber: 0,
-    xIsNext: true
-  };
-
   handleClick = i => {
-    const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const history = this.props.history.slice(0, this.props.stepNumber + 1);
     const current = history[history.length - 1];
     const squares = current.squares.slice();
     if (calculateWinner(squares) || squares[i]) {
```

Comment:
Now that you have the winner mapped in the state, you dont need to calculate the winner again.